cocos/animation/exotic-animation/exotic-animation.ts

```diff
@@ -680,6 +701,86 @@ interface ExoticTrackEvaluationRecord<TValue> {
     evaluator: ExoticTrackEvaluator<TValue>;
 }
 
+/**
+ * Exotic TRS animation graph evaluator.
+ */
+export class ExoticTrsAGEvaluation {
+    constructor (nodeAnimations: ExoticNodeAnimation[], context: AnimationClipGraphBindingContext) {
+        this._nodeEvaluations = nodeAnimations.map(
+            (nodeAnimation) => nodeAnimation.createEvaluatorForAnimationGraph(context),
+        ).filter((x) => !!x) as ExoticNodeAnimationAGEvaluation[];
+    }
+
+    public destroy () {
+        this._nodeEvaluations.forEach((nodeEvaluation) => {
+            nodeEvaluation.destroy();
+        });
+    }
+
+    public evaluate (time: number, pose: Pose) {
+        this._nodeEvaluations.forEach((nodeEvaluator) => {
```

Comment:
Avoid using `forEach` in perframe operation. This might cause garbage colloction due to the lambda function object creation.

Reply:
https://github.com/cocos/cocos-engine/pull/13503/commits/1780961fe0e8bd27cf0a5c145aecde1ef759ce97